File: front/src/types/alerts.ts
export type RiskLevel = "low" | "medium" | "high";

export type CustomerValue = "low" | "medium" | "high";

export type ChurnType = "total" | string;

export type AlertStatus = "pending" | "attended" | "dismissed";

export type HandlingChannel = "phone" | "visit" | "email" | "other";

export type InteractionResult = "positive" | "neutral" | "negative";

export type InteractionRecord = {
  id: string;
  handledBy: HandlingChannel;
  answered?: boolean;               // phone only
  visitSuccessful?: boolean;        // visit only
  emailResponseReceived?: boolean;  // email only
  result?: InteractionResult;       // only when contact was successful
  notes?: string;
  keepOpen: boolean;
  submittedAt: string;
};

export type SystemEventType = "closed" | "dismissed" | "reopened";

export type SystemEventRecord = {
  id: string;
  type: SystemEventType;
  reason?: string;
  timestamp: string;
};

export type SalesAlert = {
  id: string;
  clientName: string;
  riskLevel: RiskLevel;
  churnProbability: number;
  purchasePropensity: number;
  customerValue: CustomerValue;
  explanation: string;
  churnType: ChurnType;
  status: AlertStatus;
  interactions: InteractionRecord[];
  events: SystemEventRecord[];
  dismissReason?: string;
  dismissedAt?: string;
  alertContextJson?: string;
  predictedNextPurchase?: string;
  lastOrderDate?: string;
};

File: back/app/schemas/alerts.py
from pydantic import BaseModel


class SalesAlertResponse(BaseModel):
    id: str
    clientName: str
    riskLevel: str          # "low" | "medium" | "high"
    churnProbability: int
    purchasePropensity: int
    customerValue: str      # "low" | "medium" | "high"
    explanation: str
    churnType: str
    status: str             # "pending" | "attended" | "dismissed"
    interactions: list      # InteractionRecord[] — populated from DB
    events: list            # SystemEventRecord[] — populated from DB
    dismissReason: str | None = None
    dismissedAt: str | None = None
    alertContextJson: str | None = None
    predictedNextPurchase: str | None = None
    lastOrderDate: str | None = None
